packages/account/src/account.ts

```diff
@@ -569,6 +572,90 @@ export class Account extends AbstractAccount {
     return this.sendTransaction(request);
   }
 
+  async getTransactionCost(
+    transactionRequestLike: TransactionRequestLike,
+    { signatureCallback, quantitiesToContract = [] }: TransactionCostParams = {}
+  ): Promise<TransactionCost> {
+    const txRequestClone = clone(transactionRequestify(transactionRequestLike));
+    const isScriptTransaction = txRequestClone.type === TransactionType.Script;
+    const baseAssetId = this.provider.getBaseAssetId();
+
+    // Fund with fake UTXOs to avoid not enough funds error
+    // Getting coin quantities from amounts being transferred
+    const coinOutputsQuantities = txRequestClone.getCoinOutputsQuantities();
+    // Combining coin quantities from amounts being transferred and forwarding to contracts
+    const allQuantities = mergeQuantities(coinOutputsQuantities, quantitiesToContract);
+    // Funding transaction with fake utxos
+    txRequestClone.fundWithFakeUtxos(allQuantities, baseAssetId, this.address);
+
+    /**
+     * Estimate predicates gasUsed
+     */
+    // Remove gasLimit to avoid gasLimit when estimating predicates
+    if (isScriptTransaction) {
+      txRequestClone.gasLimit = bn(0);
+    }
+
+    /**
+     * The fake utxos added above can be from a predicate
+     * If the resources owner is a predicate,
+     * we need to populate the resources with the predicate's data
+     * so that predicate estimation can happen.
+     */
+    if (this && 'populateTransactionPredicateData' in this) {
+      (this as unknown as Predicate<[]>).populateTransactionPredicateData(txRequestClone);
+    }
```

Comment:
By funding the TX with `Account.generateFakeResources` instead of `txRequestClone.fundWithFakeUtxos`, we could remove this check enterely.

Reply:
@petertonysmith94 what was the error?

Reply:
@Torres-ssf encountered a couple issues:

- `fundWithFakeUtxos` adds an input for the `baseAssetId` (if no present), or updates the existing input [quantity](https://github.com/FuelLabs/fuels-ts/blob/3b27bac26463b39b8201db96147ec4ef091d3de0/packages/account/src/providers/transaction-request/transaction-request.ts#L596) for the estimate. I believe this is be why I'm experiencing the `InsufficientInputAmount` related errors.

- I also attempted to use the `generateFakeResources` alongside the `fundWithFakeUtxos` and removing the `populateTransactionPredicateData`. However, also encountered some issues with the correct `generateFakeResources` not being used via inheritance.

Might be beneficial for us to have a chat over huddle - as I'd like to get these breaking changes over the line ASAP.

Reply:
@petertonysmith94 Sure, I've been trying to mimic this issue without success.

Maybe it will be better to discuss it over a huddle

Reply:
@petertonysmith94 Sorry for the delay here but it took me a while to figure out what the error was.

The problem is that you forget to add an amount to fund the TX using the base asset, to cover the TX fee for the dry-run

```ts
  async getTransactionCost(
    transactionRequestLike: TransactionRequestLike,
    { signatureCallback, quantitiesToContract = [] }: TransactionCostParams = {}
  ): Promise<TransactionCost> {
    const txRequestClone = clone(transactionRequestify(transactionRequestLike));
    const baseAssetId = this.provider.getBaseAssetId();

    // Fund with fake UTXOs to avoid not enough funds error
    // Getting coin quantities from amounts being transferred
    const coinOutputsQuantities = txRequestClone.getCoinOutputsQuantities();
    // Combining coin quantities from amounts being transferred and forwarding to contracts
    const requiredQuantities = mergeQuantities(coinOutputsQuantities, quantitiesToContract);
    const resources = this.generateFakeResources(
      mergeQuantities(
        [...requiredQuantities],
        [{ assetId: baseAssetId, amount: bn('100000000000000000') }]
      )
    );

    txRequestClone.addResources(resources);

    const txCost = await this.provider.getTransactionCost(txRequestClone, {
      signatureCallback,
    });

    return {
      ...txCost,
      requiredQuantities,
    };
  }
```

Reply:
33d9b1b346cae412d901fda51a66451755477f1e